src/kaleidoscope_simulator/actions/generic_report/GenerateHostEvent.cpp

```diff
@@ -0,0 +1,394 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-Simulator -- A C++ testing API for the Kaleidoscope absolute mouse
+ * Copyright (C) 2019  noseglasses (shinynoseglasses@gmail.com)
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "kaleidoscope_simulator/actions/generic_report/GenerateHostEvent.h"
+#include "kaleidoscope_simulator/reports/BootKeyboardReport.h"
+#include "kaleidoscope_simulator/reports/KeyboardReport.h"
+#include "kaleidoscope_simulator/reports/MouseReport.h"
+#include "kaleidoscope_simulator/reports/AbsoluteMouseReport.h"
+#include "papilio/Papilio_Simulator.h"
+
+#ifdef __unix__ /* __unix__ is usually defined by compilers targeting Unix systems */
+
+#include <X11/extensions/XTest.h>
+#include <unistd.h>
+#include <linux/input-event-codes.h>
```

Comment:
fwiw, this will fail to build on any unix that isn't linux. We do currently build on the BSDs, and we should continue doing so.

Or at worst, error out on non-linux platforms that define `__unix__` with a helpful error message saying we do not support simulation there yet.